jdaviz/core/template_mixin.py

```diff
@@ -286,34 +352,16 @@ def _update_subset(self, subset, attribute=None):
             self._clear_cache("selected_obj", "selected_item")
             self._update_has_subregions()
 
-    def _selected_changed(self, event):
-        if event['new'] not in self.labels + ['']:
-            self._apply_default_selection()
-            raise ValueError(f"{event['new']} not one of {self.labels}")
-        self._clear_cache("selected_obj", "selected_item")
-        self._update_has_subregions()
-
     def _update_has_subregions(self):
         if "selected_has_subregions" in self._plugin_traitlets.keys():
             if self.selected == self._default_text:
                 self.selected_has_subregions = False
             else:
                 self.selected_has_subregions = len(self.selected_obj.subregions) > 1
 
-    @property
-    def labels(self):
-        return [s['label'] for s in self.items if 'label' in s.keys()]
-
-    @cached_property
-    def selected_item(self):
-        for item in self.items:
-            if item['label'] == self.selected:
-                return item
-        return {}
-
     @cached_property
     def selected_obj(self):
-        if self.selected == self._default_text:
+        if self.selected == self._default_text or self.selected not in self.labels:
```

Comment:
this is needed since this could be triggered by an observer in a plugin WHILE an error is raised on an invalid selection and before the default can be applied